tests/test_mc_simulator.py

```diff
@@ -0,0 +1,420 @@
+        test_dict, testing_queue
+    )
+    invoke_process_run_and_check_errors(simulator)
+
+    expected_1 = test_item[:-test_size_diff_1]
+    actual_1 = simulator.read(size=len(test_item) - test_size_diff_1)
+    assert actual_1 == expected_1
+
+    expected_2 = test_item[-test_size_diff_1:-test_size_diff_2]
+    actual_2 = simulator.read(size=test_size_diff_1 - test_size_diff_2)
+    assert actual_2 == expected_2
+
+    expected_3 = test_item[-1:]
+    actual_3 = simulator.read()
+    assert actual_3 == expected_3
+
+
+@pytest.mark.slow
+def test_MantarrayMCSimulator__handles_reads_of_size_less_than_next_packet_in_queue__when_simulator_is_running(
+    mantarray_mc_simulator,
+):
+    _, _, _, testing_queue, simulator = mantarray_mc_simulator
+
+    # remove boot up beacon
+    invoke_process_run_and_check_errors(simulator)
+    simulator.read(size=STATUS_BEACON_SIZE_BYTES)
+
+    test_item_1 = b"12345"
+    test_item_2 = b"67890"
+    test_items = [
+        {"command": "add_read_bytes", "read_bytes": test_item_1},
+        {"command": "add_read_bytes", "read_bytes": test_item_2},
+    ]
+    handle_putting_multiple_objects_into_empty_queue(test_items, testing_queue)
+
+    # Tanner (2/2/20): this try/finally block ensures that the simulator is stopped even if the test fails. Problems can arise from processes not being stopped after tests complete
+    try:
+        simulator.start()
+        confirm_queue_is_eventually_empty(testing_queue, timeout_seconds=5)
+
+        read_len_1 = 3
+        read_1 = simulator.read(size=read_len_1)
+        assert read_1 == test_item_1[:read_len_1]
+        # read remaining bytes
+        read_len_2 = len(test_item_1) + len(test_item_2) - read_len_1
+        read_2 = simulator.read(size=read_len_2)
+        assert read_2 == test_item_1[read_len_1:] + test_item_2
+    finally:
+        simulator.hard_stop()
+        simulator.join()
+
+
+def test_MantarrayMCSimulator__handles_reads_of_size_greater_than_next_packet_in_queue__when_queue_is_not_empty(
+    mantarray_mc_simulator_no_beacon,
+):
+    _, _, _, testing_queue, simulator = mantarray_mc_simulator_no_beacon
+
+    test_size_diff = 2
+    item_1 = b"item_one"
+    item_2 = b"second_item"
+
+    test_items = [
+        {"command": "add_read_bytes", "read_bytes": item_1},
+        {"command": "add_read_bytes", "read_bytes": item_2},
+    ]
+    handle_putting_multiple_objects_into_empty_queue(test_items, testing_queue)
+    invoke_process_run_and_check_errors(simulator, 2)
+    confirm_queue_is_eventually_empty(testing_queue)
+
+    expected_1 = item_1 + item_2[:test_size_diff]
+    actual_1 = simulator.read(size=len(item_1) + test_size_diff)
+    assert actual_1 == expected_1
+
+    expected_2 = item_2[test_size_diff:]
+    actual_2 = simulator.read(
+        size=len(expected_2) + 1
+    )  # reading one more byte than are present
+    assert actual_2 == expected_2
+
+    expected_3 = bytes(0)
+    actual_3 = simulator.read(
+        size=10
```

Comment:
see comment above about testing just over the edge of what would trigger the desired result.  do you see a benefit of testing 10 instead of 1 byte read size?